src/main/java/muramasa/antimatter/datagen/providers/AntimatterBlockStateProvider.java

```diff
@@ -87,13 +101,37 @@ public void state(Block block, ResourceLocation... textures) {
             simpleBlock(block, getSimpleModel(block, textures[0]));
         } else if (textures.length == 2) {
             simpleBlock(block, getLayeredModel(block, textures[0], textures[1]));
+        } else if (textures.length == 6){
+            horizontalBlock(block, getSixSidedSimpleModel(block, false, textures));
+        } else if (textures.length == 12){
+            horizontalBlock(block, getSixSidedLayeredModel(block, false, textures));
         }
     }
 
+    @Override
+    public void horizontalBlock(Block block, Function<BlockState, ModelFile> modelFunc) {
+        getVariantBuilder(block)
+                .forAllStates(state -> ConfiguredModel.builder()
+                        .modelFile(modelFunc.apply(state))
+                        .rotationY((state.has(BlockStateProperties.HORIZONTAL_FACING) ? (int) state.get(BlockStateProperties.HORIZONTAL_FACING).getHorizontalAngle() : (int) Direction.SOUTH.getHorizontalAngle()) % 360)
+                        .build()
+                );
+    }
+
```

Comment:
Again not needed, do this in your own mod in onBlockModelBuild

Reply:
Alternatively, use `BlockStateProvider#horizontalBlock`. No need to reinvent the wheel!

Reply:
removed it. the only reason I overrode it was because I was concerned that if a block did not rotate, it would crash since the block would not have that property